pandas/tests/groupby/test_groupby.py

```diff
@@ -1695,6 +1695,10 @@ def test_groupby_agg_ohlc_non_first():
         ('foo', 'sum', 'foo'))), index=pd.date_range(
             '2018-01-01', periods=2, freq='D'))
 
+    if compat.PY36:
```

Comment:
Hmm I wasn't thinking about this on original reply. Others may disagree but I don't think we want to return a different order depending on Python version. Do you know the code responsible for this behavior?

Reply:
@WillAyd What do you mean by "Do you know the code responsible for this behavior?". Do I know what is responsible for different order of columns in the DataFrame in the particular case of this test?

Reply:
@WillAyd was this question resolved?

I think that @WillAyd's question was why does the order of `result` depend on the python version?

Reply:
Yea I think this is still outstanding

Reply:
@WillAyd Your latest suggestion included this:

> When doing .agg(['ohlc', 'sum']), it passes the test in all the versions
> 
> My point was to just change the test to this instead of doing the compat. Should work in all versions then, no?

To which I responded:

> I did that in one of my commits, however I realized that the whole point of the test was to test ohlc when it's not passed in first into the agg function. There used to be an error when ohlc wasn't passed in first, but you fixed that. So the test was created to ensure that everything will still work even if ohlc isn't first, so by changing it to agg(['ohlc', 'sum']) will kind of defeat the purpose of the test, because ohlc is first. If changing it to agg(['ohlc', 'sum']) is good with you, I will do it, but I just wanted to bring this up before I do it.

Should I still change the test to .agg(['ohlc', 'sum']), which will pass in all versions? Or do you have another suggestion in mind?

Reply:
No - thanks for highlighting that about the initial purpose of the test. I guess I'm unclear as to why this change would introduce a difference in the ordering of the result across versions - shouldn't it be the same?

Reply:
@WillAyd To my understanding it should be different because previous to Python 3.6 it would sort what the user passed into the `agg` function. But since I introduced the change that would change the order of the `objs` in the `concat` function based on the Python version and `concat` is used within the .agg(['sum', 'ohlc']) call, the order of the output will be different based on the Python version (sorted in <3.6 and maintain the order that user entered in >=3.6).

Reply:
So I believe this is the offending line in the groupby code:

https://github.com/pandas-dev/pandas/blob/07625af622c650f2ea124e3ddc5a80df37e6a2aa/pandas/core/groupby/generic.py#L788

The dict which gets passed into that concat call is built here:

https://github.com/pandas-dev/pandas/blob/07625af622c650f2ea124e3ddc5a80df37e6a2aa/pandas/core/groupby/generic.py#L825

I'm not sure of best approach to handle. One one hand maybe we convert all of the groupby code to OrderedDict to prevent this, but at the same time I'm not sure why this would appear as part of this PR if the focus is only on OrderedDict, which isn't in use here.

Any idea where that impact may be stemming from within this?

Reply:
@WillAyd This PR also deals with regular dicts in Python 3.6+ because they should exhibit the same behaviour as OrderedDict in Python 3.6+. So this PR ensures that order of user input in the concat function is maintained for OrderedDicts in all versions and regular dicts in Python 3.6+. That's why regular dicts are also affected based on the version. With this in mind, do you think that we should convert the dict that you specified in the above comment to an OrderedDict for the result to be the same in all versions?

Reply:
@alexander-ponomaroff if you change the ``results = OrderedDict()`` in groupby/generic.py as above. you won't need to change this test code.

Reply:
@alexander-ponomaroff can you update?

I don't think that this PR should require changing the test. Rather, we'll update groupby to ensure that the columns are consistent with the previous results for all versions of Python.